tests/infrastructure/programs/sre/test_firewall.py

```diff
@@ -0,0 +1,175 @@
+from collections.abc import Callable
+from enum import Enum
+from functools import partial
+
+import pulumi
+import pulumi.runtime
+import pytest
+from pulumi_azure_native import network
+
+from data_safe_haven.infrastructure.programs.sre.firewall import (
+    SREFirewallComponent,
+    SREFirewallProps,
+)
+
+from ..resource_assertions import assert_equal
+
+
+class MyMocks(pulumi.runtime.Mocks):
+    def new_resource(self, args: pulumi.runtime.MockResourceArgs):
+        resources = [args.name + "_id", args.inputs]
+        return resources
+
+    def call(self, _: pulumi.runtime.MockCallArgs):
+        return {}
+
+
+# TODO: These breaks many other tests!
+pulumi.runtime.set_mocks(
+    MyMocks(),
+    preview=False,
+)
```

Comment:
Luckily, you don't need these :). See the `test_application_gateway.py` for an example of Pulumi runtime tests without `set_mocks`.